### ISSUE TITLE
compile error in utf8 unittest

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#4861]
I am getting an error while building the unittests from `next`. The server works as expected. The specific error is:
```
src/unittest/utf8.cc:573:124: error: expression with side effects has no effect in an unevaluated context [-Werror,-Wunevaluated-expression]
        switch (0) case 0: default: if (const ::testing::AssertionResult gtest_ar = (::testing::internal:: EqHelper<(sizeof(::testing::internal::IsNullLiteralHelper(*current++)) == 1)>::Compare("*current++", "*it++", *current++, *it++))) ; else return ::testing::internal::AssertHelper(::testing::TestPartResult::kFatalFailure, "src/unittest/utf8.cc", 573, gtest_ar.failure_message()) = ::testing::Message();
```

This is a little odd, since there has been no change in that file since August 10.

**Comment by danielmewes** [comment]
Which platform / compiler is this with?

**Comment by larkost** [comment]
Sorry... meant to write that once my compile on `newton` finished. This happens reliably on my Mac, but I am not able to reproduce on `gantz`. There was just an Xcode update today which I did locally, which might account for it.

**Comment by larkost** [comment]
I updated `gantz` to the lasted OS X version (10.10.5), and updated Xcode and that causes this problem there too. So it looks like that is what is going on here.

**Comment by danielmewes** [comment]
Fixed in `next` 476d2e6 and also cherry-picked into `v2.1.x` with d6cad7a.
